core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -300,6 +349,22 @@ private[spark] class SparkSubmitArguments(args: Seq[String], env: Map[String, St
         propertiesFile = value
         parse(tail)
 
+      case ("--kill") :: value :: tail =>
```

Comment:
as discussed offline, we will continue to use the `--kill` semantics here for now.